From 2695834b9ca7b028515c4a69d7cf58dc45338af8 Mon Sep 17 00:00:00 2001
From: Fabien Dessenne <fabien.dessenne@foss.st.com>
Date: Wed, 1 Dec 2021 17:39:03 +0100
Subject: [PATCH] arm64: dts: st: add IPCC1 node on stm32mp251

Define the IPCC1 node to support the IPCC mailbox for communication
between the Cortex-A35 and Cortex-M33

Signed-off-by: Fabien Dessenne <fabien.dessenne@foss.st.com>
Change-Id: I37a83656e0783965289f3a155b91879a9448a306
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 87e990e1e44e..fa34d3c298cd 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -303,6 +303,18 @@ hpdma3: dma-controller@40420000 {
 			st,axi-max-burst-len = <16>;
 		};
 
+		ipcc1: mailbox@40490000 {
+			compatible = "st,stm32mp1-ipcc";
+			#mbox-cells = <1>;
+			reg = <0x40490000 0x400>;
+			st,proc-id = <0>;
+			interrupts = <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 172 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "rx", "tx";
+			clocks = <&scmi_clk CK_BUS_IPCC1>;
+			status = "disabled";
+		};
+
 		ommanager: ommanager@40500000 {
 			#address-cells = <2>;
 			#size-cells = <1>;
-- 
2.39.5

